packages/@react-spectrum/button/src/Button.tsx

```diff
@@ -39,15 +44,44 @@ function Button<T extends ElementType = 'button'>(props: SpectrumButtonProps<T>,
     style = variant === 'accent' || variant === 'cta' ? 'fill' : 'outline',
     staticColor,
     isDisabled,
+    isPending,
     autoFocus,
     ...otherProps
   } = props;
   let domRef = useFocusableRef(ref);
   let {buttonProps, isPressed} = useButton(props, domRef);
   let {hoverProps, isHovered} = useHover({isDisabled});
+  let stringFormatter = useLocalizedStringFormatter(intlMessages);
   let {styleProps} = useStyleProps(otherProps);
+  let progressCircleProps: SpectrumProgressCircleProps = {
+    'aria-label': stringFormatter.format('loading'),
+    isIndeterminate: true,
+    size: 'S',
+    ...(
+      (variant === 'overBackground' && style !== 'fill') ||
+      ((variant === 'cta' || variant === 'accent') && style !== 'outline')
+    ) && {variant: 'overBackground'}
+  };
   let hasLabel = useHasChild(`.${styles['spectrum-Button-label']}`, domRef);
   let hasIcon = useHasChild(`.${styles['spectrum-Icon']}`, domRef);
+  let timeout = useRef<ReturnType<typeof setTimeout>>();
+  let [showLoader, setShowLoader] = useState(false);
+
+  useEffect(() => {
+    if (isPending) {
+      // Start timer when isPending is set to true.
+      timeout.current = setTimeout(() => {
```

Comment:
I think you might not need a ref for this? A local variable within the effect should work since the cleanup function has access to it.

Reply:
Thanks. That was a holdover from Dylan's PR. Since we don't need to pass the ref around, it isn't needed.